composables/drop/massmint/useDropMassMint.ts

```diff
@@ -0,0 +1,265 @@
+import {
+  AllocatedNFT,
+  DoResult,
+  allocateCollection,
+  allocateClaim as claimAllocation,
+} from '@/services/fxart'
+import { EntropyRange, GenerativePreviewItem } from '../useGenerativePreview'
+import { ImageDataPayload } from '../useGenerativeIframeData'
+import { ToMintNft } from '@/components/collection/drop/types'
+import { getFakeEmail } from '@/components/collection/drop/utils'
+import { TransactionStatus } from '../../useTransactionStatus'
+import useDropMassMintPreview from './useDropMassMintPreview'
+import useDropMassMintUploader from './useDropMassMintUploader'
+import useDropMassMintListing from './useDropMassMintListing'
+import useDropMassMintRefs from './useDropMassMintRefs'
+import { useCollectionEntity } from '../useGenerativeDropMint'
+
+export type MassMintNFT = ToMintNft & {
+  imageDataPayload?: ImageDataPayload
+  metadata?: string
+  hash: string
+  sn?: number
+  entropyRange: EntropyRange
+}
+
+type MassMintParams = {
+  isLoading: Ref<boolean>
+  status: Ref<TransactionStatus>
+  isError: Ref<boolean>
+  isTransactionLoading: Ref<boolean>
+  onSubmitMints: () => Promise<void>
+  onTransactionCancel?: () => void
+}
+
+export default ({
+  isLoading,
+  isTransactionLoading,
+  isError,
+  status,
+  onSubmitMints,
+  onTransactionCancel,
+}: MassMintParams) => {
```

Comment:
why?

composables are suppose to be easy to use anywhere
all thee arguments make it not easy

it looks to me that you are composing differant composables in one BIG one

why?

keep them simple and easy to use, do the composition in the component

especially since this is very unlikely to be used in many places. it is just for 1 component